pkgs/applications/editors/vscode/extensions/shader-slang.slang-language-extension/default.nix

```
@@ -0,0 +1,79 @@
+{
+  lib,
+  shader-slang,
+  stdenv,
+  vscode-utils,
+}:
+
+vscode-utils.buildVscodeMarketplaceExtension {
+  mktplcRef = {
+    name = "slang-language-extension";
+    publisher = "shader-slang";
+    version = "2.0.1";
+    hash = "sha256-cMBFksLcEi3OmRpyuk/hMNo5HqsfGvWhheiLg6Dusn0=";
+  };
+
+  postInstall =
+    let
+      # https://github.com/shader-slang/slang-vscode-extension/blob/v2.0.1/build_scripts/build-package.sh#L40-L43
+      binaries = {
+        "x86_64-linux" = {
+          name = "linux-x64";
+          files = [
+            "lib/libslang.so"
+            "lib/libslang-glsl-module.so"
+            "bin/slangd"
+          ];
+        };
+        "aarch64-linux" = {
+          name = "linux-arm64";
+          files = [
+            "lib/libslang.so"
+            "lib/libslang-glsl-module.so"
+            "bin/slangd"
+          ];
+        };
+        "x86_64-darwin" = {
+          name = "darwin-x64";
+          files = [
+            "lib/libslang.dylib"
+            "lib/libslang-glsl-module.dylib"
+            "bin/slangd"
+          ];
+        };
+        "aarch64-darwin" = {
+          name = "darwin-arm64";
+          files = [
+            "lib/libslang.dylib"
+            "lib/libslang-glsl-module.dylib"
+            "bin/slangd"
+          ];
+        };
+      };
+    in
+    lib.strings.concatLines (
+      [
+        "bin=$out/share/vscode/extensions/shader-slang.slang-language-extension/server/bin"
+        "rm -r $bin"
+      ]
+      ++ (
+        if builtins.hasAttr stdenv.system binaries then
+          with binaries.${stdenv.system};
+          [ "mkdir -p $bin/${name}" ] ++ (map (path: "cp ${shader-slang}/${path} $bin/${name}/") files)
+        else
+          [ ]
+      )
+    );
```

Continuing from above. 
```suggestion
    lib.concatLines [
        "bin=$out/share/vscode/extensions/shader-slang.slang-language-extension/server/bin"
        "rm -r $bin"
       ("mkdir -p $bin/${name}" ] ++ (map (path: "cp ${shader-slang}/${path} $bin/${name}/") files))
     ];
```